project/BuildHelper.scala

```diff
@@ -76,55 +85,30 @@ object BuildHelper {
           "-Ywarn-nullary-unit",
           "-Xexperimental",
           "-Ywarn-unused-import"
-        ) ++ std2xOptions
+        )
       case _ => Seq.empty
     }
 
+    stdOptions ++ extraOptions ++ optimizerOptions
+  }
+
+  val buildInfoSettings = Seq(
+    buildInfoKeys := Seq[BuildInfoKey](name, version, scalaVersion, sbtVersion, isSnapshot),
+    buildInfoPackage := "zio",
+    buildInfoObject := "BuildInfoZioMacros"
+  )
+
   def stdSettings(prjName: String) =
     Seq(
       name := s"$prjName",
-      scalacOptions := stdOptions,
       crossScalaVersions := Seq("2.13.0", "2.12.8", "2.11.12"),
       scalaVersion in ThisBuild := crossScalaVersions.value.head,
-      scalacOptions := stdOptions ++ extraOptions(scalaVersion.value),
-      libraryDependencies ++= compileOnlyDeps ++ testDeps ++ Seq(
-        compilerPlugin("org.typelevel"   %% "kind-projector"  % "0.10.3"),
-        compilerPlugin("com.github.ghik" %% "silencer-plugin" % "1.4.2")
-      ) ++ {
-        CrossVersion.partialVersion(scalaVersion.value) match {
-          case Some((2, x)) if x >= 13 =>
-            Nil
-          case _ =>
-            Seq(compilerPlugin(("org.scalamacros" % "paradise" % "2.1.1").cross(CrossVersion.full)))
-        }
-      },
+      scalacOptions := compilerOptions(scalaVersion.value, optimize = !isSnapshot.value),
```

Comment:
shouldn't it be `optimize = isSnapshot.value` ?
also @mijicd is `isSnapshot` going to be set with current release config?

I'd like to make sure when releasing optimalizations are enabled

Reply:
We don't publish snapshot at the moment, but might support them in the future. However, take a look at [this PR](https://github.com/xerial/sbt-sonatype/pull/96) as it addresses an issue with `isSnapshot` in `sbt-sonatype`.

Reply:
@mijicd very interesting read, thanks!

@ioleo Depends on when we want optimization to run. The way it is right now only builds that are on a tagged commit and in a non-dirty repository will be optimized. I believe that is the right way to go about it